### PR TITLE
E2E tests: Slack notifications

### PR DESCRIPTION
**Pull request by adimoldovan**
#### Changes proposed in this Pull Request:

This PR improves the Slack notifications for E2E tests. The scope was to produce less noise while enhancing the provided information for a quick evaluation.

Changes:
* Send notification for each failed test run (matrix group).
* Send a single notification for each successful E2E workflow run. This is more as a heartbeat notification, in case there are no test failures we still want to know the setup still works fine.
* Included the test suite name.
* Included list of failed tests.
* Build notification content out of Jest's test summary json file instead of custom slack log file.
* Log files are no longer uploaded in Slack, only stacktraces and screenshots are, for a quick evaluation. In case more details are needed we can check the logs together with the full test output as they are attached in Github.
* General look and feel.

#### Jetpack product discussion
1200222342137241-as-1200182606016954

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:

* Locally, you can set `GITHUB_EVENT_PATH` and `GITHUB_RUN_ID` env variables and run `./bin/slack.js` as per instructions.
* In CI, check the notifications channel (example: p1620640477371800-slack-C1Y1F7ZM1)

### CODE REVIEW

File: projects/plugins/jetpack/tests/e2e/bin/slack.js [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+
+	const detailLines = [];
+	const failureDetails = [];
+	const screenshots = fs.readdirSync( config.get( 'dirs.screenshots' ) );
+	const matchedScreenshots = [];
+
+	// Go through all test results and extract failure details
+	for ( const tr of result.testResults ) {
+		for ( const ar of tr.assertionResults ) {
+			if ( ar.status !== 'passed' ) {
```
Comment (brbrr) [inline projects/plugins/jetpack/tests/e2e/bin/slack.js:76]: sending log files to the slack could be useful when looking through failures in multiple runs

**Review by brbrr — CHANGES_REQUESTED** [review]
It looks great! 

Let's add log files to the failed reports as we discussed during the call. 

Also, I noticed you get rid of `GUTENBERG` env variable. Not sure if that was intentional, but right now we wouldn't be able to identify tests with the latest Gutenberg. (probably that's fine)

File: projects/plugins/jetpack/tests/e2e/bin/slack.js (hunk 1 above)
Comment (adimoldovan) [reply projects/plugins/jetpack/tests/e2e/bin/slack.js:76]: Done

File: projects/plugins/jetpack/tests/e2e/bin/slack.js [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+	let result;
+
+	// If results summary file is not found send failure notification and exit
+	try {
+		result = JSON.parse( fs.readFileSync( 'output/summary.json', 'utf8' ) );
+	} catch ( error ) {
+		const errMsg = 'There was a problem parsing the test results file.';
+		console.error( errMsg );
+		await sendMessage( buildDefaultMessage( false, errMsg ), {} );
+		return;
```
[the remark this answers is not in the document]
Comment (adimoldovan) [reply projects/plugins/jetpack/tests/e2e/bin/slack.js:65]: I don't think `throw` is the right choice. The desired outcome here is to send a notification with the message from above line, not to fail. The result should be success.

File: projects/plugins/jetpack/tests/e2e/bin/slack.js [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+				describe: 'Job status',
+				type: 'string',
+			} );
+		},
+		async argv => await reportJobRun( argv.status )
+	)
+	.help( 'h' )
+	.alias( 'h', 'help' ).argv;
+
+//endregion
```
[the remark this answers is not in the document]
Comment (adimoldovan) [reply projects/plugins/jetpack/tests/e2e/bin/slack.js:45]: These are used to group code under foldable regions, so they can be collapsed in the IDE. It helps me navigate larger files. This syntax is by default supported in WebStorm, not sure about other IDEs.

**Comment by adimoldovan** [comment]
> I noticed you get rid of `GUTENBERG` env variable.

The group name is included in the notification and we know that `gutenberg` group uses latest in CI.

**Review by brbrr — APPROVED** [review]
Looks great!

**Pull request merged**